README.md

```diff
@@ -178,6 +178,26 @@ The "Delete Staging Deployment" action allows you to delete the deployment not r
           app-name: <app name>
 ```
 
+## Arguments
+> [!NOTE]
+> Some arguments are only applicable for certain settings of the `action` argument. The Action column below specifies the pertinent actions for each argument. Any argument listed as Required is only required for the pertinent Action(s).
+
+|Argument|<div style="width:100px">Action</div>|Required|Description|
+|--- |--- |--- |--- |
+|`action`|all|Required| The action to be performed by this task.<br/>One of: `deploy`, `set-production`, `delete-staging-deployment`<br/>Default value: `deploy`|
+|`azure-subscription`|all|Required| The Azure subscription ID for the target Azure Spring Cloud instance.|
+|`service-name`|all|Required| The name of the Azure Spring Cloud service instance.|
+|`app-name`|all|Required| The name of the Azure Spring Cloud app to deploy. The app must exist prior to task execution.
+|`use-staging-deployment`|deploy<br/>set-production|Optional| If set to `true`, apply the task to whichever deployment is set as the staging deployment at time of execution. If set to `false`, apply the task to the production deployment.<br/>Default value: `true`|
+|`deployment-name`|deploy<br/>set-production|Optional| The name of the deployment to which the action will apply. It overrides the setting of `use-staging-deployment`.|
+|`create-new-deployment`|deploy|Optional| If set to `true` and the deployment specified by `deployment-name` does not exist at execution time, it will be created.<br/>Default value: `false`|
+|`package`|deploy|Required| The file path to the package containing the application to be deployed (`.jar` file for Java, `.zip` for .NET Core) or to a folder containing the application source to be built. <br/>Default value: ```${{ github.workspace }}/**/*.jar```|
```

Comment:
please move this up to after app-name so that all the required are listed first. others lgtm

Reply:
As the discription above the table, this is not required of all actions. This only required in deploy action. The order follows pipeline task doc. So I am not sure if it should be placed after app-name.